lib/Conversion/TorchToLinalg/IndirectDataMovement.cpp

```diff
@@ -168,6 +168,200 @@ class ConvertAtenEmbeddingOp : public OpConversionPattern<AtenEmbeddingOp> {
 };
 } // namespace
 
+namespace {
+// AtenEmbeddingPaddingIdxOp
+// SUM mode == integer 0
+// Sums bags of embeddings together from a weight tensor based on an index and offset Vector.
+// Example arguments
+// weight = [[1, 3, 5, 3],                <- contains 4 embedding each of length 4
+//           [3, 4, 2, 1],
+//           [2, 2, 3, 2],
+//           [0, 4, 2, 1]]
+//
+// indices = [0, 2, 3, 1, 2, 3, 2, 1, 0, 1] <- always 1D Tensor
+//
+// offsets = [0, 3, 5]                      <- always 1D Tensor
+//
+// output_tensor = initZeroTensor(offsets_length, embedding_size)
+//
+// for i in range(offsets_length):         <- dim0
+//     for j in range(indices_length):     <- dim1 
+//         for k in range(embedding_size): <- dim2 
+//             if(offsets[i] <= j and j < offsets[i+1]):
+//                 output_tensor[i][k] = output_tensor[i][k] + weight[indices[j]][k]
+//             else:
+//                 break
+//
+// Indexing maps for linalg::Generic ops
+//          
+// 
+// indices_indexing_map  = (d0, d1, d2) -> (d1)
+// offset_indexing_map   = (d0, d1, d2) -> (d0)
+// output_indexing_map   = (d0, d1, d2) -> (d0, d2)
+//
+
+class ConvertAtenEmbeddingBagPaddingIdxOp : public OpConversionPattern<AtenEmbeddingBagPaddingIdxOp> {
+public:
+  using OpConversionPattern::OpConversionPattern;
+  LogicalResult
+  matchAndRewrite(AtenEmbeddingBagPaddingIdxOp op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+    if (failed(verifyLinalgCompatibleTypes(op, rewriter)))
+        return failure();
+    Location loc  = op->getLoc();
+    auto context = op->getContext();
+    Value weight  = adaptor.weight();
+    Value indices = adaptor.indices();
+    Value offsets = adaptor.offsets();
+    Value scale_grad_by_freq = adaptor.scale_grad_by_freq();
+    Value mode = op.mode();
+    Value sparse = adaptor.sparse();
+    Value include_last_offset = adaptor.include_last_offset();
+
+    int64_t m;
+    //TODO: handle mean and max modes
+    //sum mode
+    if (!matchPattern(mode, m_TorchConstantInt(&m))){
+     return rewriter.notifyMatchFailure(op, "mode is expected to be an integer value");
+    }  
+    
+    //get the Embedding dimension
+    Value embeddingDim = getDimOp(rewriter, loc, weight, 1);
```

Comment:
This is only used to calculate the output tensor size below, so I would move it to be next to that calculation